icu4c/source/common/cmemory.h

```diff
@@ -725,9 +725,14 @@ class MemoryPool : public UMemory {
     }
 
     MemoryPool& operator=(MemoryPool&& other) U_NOEXCEPT {
-        fCount = other.fCount;
-        fPool = std::move(other.fPool);
-        other.fCount = 0;
+        // Since `this` may contain instances that need to be deleted, we can't
+        // just throw them away and replace them with `other`. The normal way of
+        // dealing with this in C++ is to swap `this` and `other`, rather than
+        // simply overwrite: the destruction of `other` can then take care of
+        // running MemoryPool::~MemoryPool() over the still-to-be-deallocated
+        // instances.
```

Comment:
Excessive documentation for the codebase? I could have simply given this explanation in the code review itself.

Reply:
I think this is a place where a lot of documentation is good to have. It would probably have avoided confusion if the current version of this code had explicitly spelled out that the object being assigned to was expected to be empty.